### DISCUSSION TITLE
工程師會不會被取代呢?

### DISCUSSION THREAD

**Discussion by wangtyphoon** [1101-GS4719-C/Python-Community-GS4719-C#908] [General]
近年來ai 的發展主要是由於硬體技術進步所驅動，算力無疑是靠現金堆積出來的，日前曾經研發出圍棋ai Alphago的工師，向工程師發起挑戰!推出alphacode 登上期刊Science的封面頭條，已經能讚特定領域達到工程師的平均水平，究竟工程師會不會被自己創造出來的AI給取代呢?
https://www.ithome.com.tw/news/149181

**Comment by ching020721** [comment]
好像有點厲害...只能說人類在AI進步的同時也應當持續精進，才不會面臨被淘汰的境界

**Comment by manipulateed** [comment]
看到這篇讓我開始思考，還有什麼是ai 無法替代的東西，他能訓練人的思維能統計大量的演算法與代碼，留給未來的工作機會感覺正在慢慢流失。

**Comment by HuangHsinTzu** [comment]
現在很多人都因為面臨資訊時代而許選擇學習程式設計
現在連AI都會coding了嗎?!
我認為我們在學習學習程式設計的同時也要培養人文素養
要能發揮創意靈活應用程式
因為若只是學習程式能力，就很難獨樹一幟，因為永遠會有人(甚至是AI)比你厲害

**Comment by Kotori0501** [comment]
看到這篇文章，感覺陸陸續續的工作都快被ai取代了，說不定未來會從以前的人與人的交流變成人與機器人的交流，想想就有點可怕

**Comment by peggy0329** [comment]
這篇文章讓我想到未來的世界會不會像電影「瓦力」的世界一樣
所有工作、生活都充滿了AI智慧與機器人
人們的工作除了被AI取代外，也因為高科技使人們變得怠惰

我認為我們除了具備基本的程式能力，也要培養第二專長
同時多方涉取多元知識，讓自己變得更加彈性
才能避免自己未來成為只能仰賴AI技術的人類

**Comment by Evan0979** [comment]
之後我們連學習的速度都比不上AI了,想到我花超久時間讀的書,AI可能3秒就學完然後還做得比我更好,好想哭啊QQ

**Comment by ruthchen1020** [comment]
我覺得如果工程師的等級太低應該會被取代，畢竟低階的程式語言可以用簡單的邏輯去生成，但高階的程式語言我覺得沒有那麼簡單，因為通常沒有一個標準寫法，而且會依照各種情況去調整邏輯。但AI可以幫忙最佳化程式語言，畢竟用大數據最佳化才是AI的強項。

**Comment by StevenWanglolz** [comment]
我認為AI應該是沒辦法取代好的工程師，因為一個厲害的工程師的能力不只有接收指令和生出程式碼這麼簡單!

**Comment by CPCpyt** [comment]
我想比起技術本身，工程師的創作邏輯以及創新運用更為重要。
而這些恰好是無法被複製模仿的。

**Comment by swerfg03joe** [comment]
與其說工程師會被AI取代，不如說是該工程師能力不足無法想出新的方法，AI再怎麼優秀他也只能根據已存在的技術去生成code，無法憑空生出全新的事物，所以人應蓋將AI視為是一種輔助學習的工具才對，既然他寫出的程式碼總是比你優秀、簡潔。那其實你也可以將它的風格學起來，價值是自己去創造的不是別人賦予你的。

**Comment by ru0372572** [comment]
但AI學了那麼多，我們所想的那些所謂創新，在AI的眼中說不定是幾年前就在別的地方學過的東西，我們也不是了解世界上所有的技術，但他們能夠彙整龐大的數據並且比人類蒐集更完整的文獻。

**Comment by BlahFish1006** [comment]
想到現在我在學習程式語言的時候，常常也是看ChatGPT的整理，有種演算法在教我演算法的詭異感覺......
想要不被取代真的好難啊，對現代人的技能水準要求感覺越來越嚴苛了。

**Comment by WEI-CHEN-CHUN** [comment]
科技始終來自人性，人類若不想辦法精進，連思考創意都會被取代。

**Comment by wenchi9023** [comment]
AI現在已經取代某些較routine的工作了，未來很難說
前幾天才和實習主管聊到，他說現代人都覺得已經學夠多，但這些都只是在趕上世界變化速度而已，真的還有好多要學習和要會運用...

**Comment by Pietra1226** [comment]
我個人覺得是不會被取代，因為設計還是需要想像力，雖然無法設計出效能最好的樣品，但可以想出不同的觀點，再讓AI去實作

**Comment by s96015welson** [comment]
我認為工程師在程式設計方面有獨創性，技術門檻也較高，因此不會簡單被取代，而ai應該是扮演輔助工程師的角色。

**Comment by Jzd314** [comment]
工程師感覺很難被AI取代，AI擁有的東西基本上還是來自工程師，但AI能夠具有很多個工程師的資訊，因此工程師未來可能會變成輔助AI的角色，但無法被完全取代

**Comment by woohsin** [comment]
AI的發展確實受到硬體技術進步的推動，算力的提升是AI進步的關鍵因素之一。然而，AI的發展不僅僅侷限於硬體，軟體演算法、數據量和質量、以及對領域知識的理解也同樣重要。目前的AI技術雖然在某些特定任務上已經能達到甚至超越人類水平，但要取代工程師還有相當長的路要走。

工程師的工作不僅僅是執行特定任務，更包括問題解決、創意發揮和對系統的整體理解等方面。AI目前還無法完全模擬人類的智慧和靈活性，尤其是在面對複雜、不確定的情況下。因此，儘管AI在某些方面已經顯示出了驚人的能力，但工程師仍然在AI發展和應用的過程中扮演著不可或缺的角色。AI更多地是作為工程師的輔助工具，幫助他們更高效地完成任務，而非取代他們。

**Comment by jylin34** [comment]
我認為不會。軟體工程是一門解決複雜問題的科學，在實務中，每天都會遇到從未見過的問題，這需要工程師將複雜問題拆解成一個個簡單的子問題，而AI只能在某些碎片化的子問題上提供幫助。除此之外，我認為工程師被認為會被AI取代的原因在於這是一個具有標竿性和挑戰性的職業。如果工程師被取代，許多其他工作機會也將隨之消失。

**Comment by titan92601** [comment]
現在大多數認同AI的進步確實受益於硬體技術的發展，算力對於AI的演算和訓練至關重要。Alphacode的出現顯示了AI在特定領域已經能與工程師媲美。但我認圍工程師不會被AI完全取代。AI雖然能處理特定任務，但仍需人類設計、監督和優化。工程師在創造和引導AI發展中扮演著不可或缺的角色，而人類的創造力和解決問題能力是AI所無法取代

**Comment by NcuJerryKao** [comment]
提升自身技能：工程師應該持續學習和提升技能，特別是在AI相關的領域。掌握最新的技術和工具，保持競爭力。例如，了解機器學習、深度學習和數據科學等領域的最新進展。

跨領域合作：與其他領域的專家合作，發展跨學科的知識和技能。這樣可以提供更多創新和解決問題的方法，增加職場價值。

專注於創意和人際互動：AI在創意和人際互動方面仍存在限制。工程師應該發展這些方面的能力，因為這些是目前AI難以完全取代的。

參與AI發展：積極參與AI的研發和應用，成為技術進步的一部分，而不是被動的接受者。這樣可以確保自身在未來的技術變革中保持領先地位。

道德與責任意識：關注AI技術的道德和社會影響，確保其發展對社會有正面影響。工程師應該參與制定和推動AI技術的倫理準則和法律法規。

**Comment by alexender0104** [comment]
工程師在當前和未來仍然是非常重要的角色，但隨著技術的發展和自動化的普及，工程師可能會面臨某些工作上的變化。我認為某些重複性高且簡單的工程任務可能被自動化技術取代，例如自動化測試、代碼生成和檢查，以及隨著新興技術的發展，如物聯網（IoT）、區塊鏈和人工智能，對於相應技術領域的專業工程師需求會增加。

**Comment by starshatters** [comment]
我覺得工程師面臨AI技術進步的挑戰，的確是當前和未來需要思考的重要議題。AI的發展主要受惠於硬體技術的進步，尤其是算力的提升，使得更複雜和智能的算法得以實現。然而，工程師與AI之間並非零和遊戲。AI在特定領域能夠達到甚至超越工程師的水平，這些領域通常是需要大量數據處理或複雜算法應用的場景。但是，AI無法替代工程師在許多其他重要方面的角色，例如創新、問題解決、系統設計和人際交流等能力。

**Comment by JustAHedgehog** [comment]
我相信工程師是不會被取代的，但會經歷篩選，留下能繼續歲時代進步的工程師，像之前在聽「人工智慧發展趨勢與倫理議題面臨的挑戰」講座時就有提到即是主管也需要 ChatGPT 幫忙撰寫一些簡單的code以加快進度，就像以前公務員做的事情只有蓋印章而已，現在還要會打word excel 操作機器等等技能，當初可是裁掉多少員工啊，但現在我們都認為這些技能是基操，所以現在的現象只是某種程度的歷史重演而已

**Comment by hbh10601** [comment]
工程師不會完全被取代，但會留下菁英，像之前黃爸爸說的，人類反而可能會回去鑽研農業等等的，科技方面可能就會交由AI去執行

**Comment by tshsu04224** [comment]
個人認為現在AI這類的技術還需要由許多工程師來維護或優化，甚至可能產出更多更棒的工具，因此現階段應該不太容易被取代，在許多部分上都還需要工程師精進這些新技術，只不過低階一些的工程師可能就會被取代，畢竟AI編寫程式碼真的十分快速，準確度也有一定的水準

**Comment by sandyho0307** [comment]
我覺得未來可能會是人機協作，而不是完全取代。AI可以幫助工程師提高效率，完成一些重複性高、需要大量計算的工作，而工程師則可以專注於更高層次的設計和創新工作。

**Comment by Shang621** [comment]
工程師不太可能完全被取代，但他們的工作內容將隨著科技進步而改變。人工智能和自動化可以處理重複性任務，使工程師能專注於創新和複雜問題解決。未來，工程師需要具備跨領域知識和創造力，以適應新技術和挑戰。
